rust/crd/src/security.rs

```diff
@@ -285,49 +249,40 @@ impl KafkaTlsSecurity {
     pub fn add_volume_and_volume_mounts(
         &self,
         pod_builder: &mut PodBuilder,
-        cb_prepare: &mut ContainerBuilder,
         cb_kcat_prober: &mut ContainerBuilder,
         cb_kafka: &mut ContainerBuilder,
     ) {
         // add tls (server or client authentication volumes) if required
         if let Some(tls_server_secret_class) = self.get_tls_secret_class() {
-            cb_prepare.add_volume_mount("server-tls-mount", Self::STACKABLE_TLS_SERVER_MOUNT_DIR);
-            // kcat requires pem files and not keystores
-            cb_kcat_prober
-                .add_volume_mount("server-tls-mount", Self::STACKABLE_TLS_SERVER_MOUNT_DIR);
-            cb_kafka.add_volume_mount("server-tls-mount", Self::STACKABLE_TLS_SERVER_MOUNT_DIR);
+            // We have to mount tls pem files for kcat (the mount can be used directly)
+            cb_kcat_prober.add_volume_mount(
+                Self::STACKABLE_TLS_CERT_SERVER_DIR_NAME,
+                Self::STACKABLE_TLS_CERT_SERVER_DIR,
+            );
             pod_builder.add_volume(Self::create_tls_volume(
-                "server-tls-mount",
+                Self::STACKABLE_TLS_CERT_SERVER_DIR_NAME,
                 tls_server_secret_class,
             ));
-
-            // empty mount for trust and keystore
-            cb_prepare.add_volume_mount("server-tls", Self::STACKABLE_TLS_SERVER_DIR);
-            cb_kafka.add_volume_mount("server-tls", Self::STACKABLE_TLS_SERVER_DIR);
-            pod_builder.add_volume(
-                VolumeBuilder::new("server-tls")
-                    .with_empty_dir(Some(""), None)
-                    .build(),
+            // Mounted keystores fore the kafka container
+            cb_kafka.add_volume_mount(
+                Self::STACKABLE_TLS_KEYSTORE_SERVER_DIR_NAME,
+                Self::STACKABLE_TLS_KEYSTORE_SERVER_DIR,
             );
+            pod_builder.add_volume(Self::create_tls_keystore_volume(
```

Comment:
Nit: I would prefer to code to first add the volume and the mount afterwards "to tell a story", but that's just an idea

Reply:
https://github.com/stackabletech/kafka-operator/pull/621/commits/8a1f1acbad000145fc65ca6b43ee49a6b58a9582